### PR TITLE
feat(ARCH-349/forms): lazyload react ace

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

react-ace is downloaded (part of dependencies.json manifest) while it will may be never used.

**What is the chosen solution to this problem?**

POC a way to lazy load UMD.

**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/importFromCDN.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+			script.setAttribute('type', 'text/javascript');
+			document.body.appendChild(script);
+		}
+		if (varName) {
+			const intervalId = setInterval(() => {
+				if (window[varName]) {
+					clearInterval(intervalId);
+					resolve(window[varName]);
+				}
+			}, 200);
```
Comment (frassinier) [inline packages/components/src/importFromCDN.js:23]: I'm not confident there is no error management at this point

File: packages/forms/src/deprecated/widgets/CodeWidget/__snapshots__/CodeWidget.test.js.snap [hunk 2]
```diff
@@ -1,48 +1,24 @@
...
-      0,
-    ]
-  }
-  setOptions={
-    Object {
-      "enableBasicAutocompletion": true,
-      "enableLiveAutocompletion": true,
-      "enableSnippets": true,
+<AceCodeWidget>
+  <UNDEFINED
```
Comment (frassinier) [inline packages/forms/src/deprecated/widgets/CodeWidget/__snapshots__/CodeWidget.test.js.snap:5]: Can we add a displayName?

File: packages/forms/src/deprecated/widgets/CodeWidget/CodeWidget.component.js [hunk 3]
```diff
@@ -45,25 +54,27 @@ try {
...
-					setOptions={setOptions}
-					enableSnippets
-					value={value}
-					theme="chrome"
-					showGutter={false}
-					showPrintMargin={false}
-					{...contextProps}
-					{...options}
-				/>
+				<Suspense fallback={<div>Loading...</div>}>
```
Comment (frassinier) [inline packages/forms/src/deprecated/widgets/CodeWidget/CodeWidget.component.js:57]: not i18n ready, may I suggest to  use a skeleton here?

File: packages/forms/src/deprecated/widgets/CodeWidget/CodeWidget.component.js [hunk 4]
```diff
@@ -1,12 +1,21 @@
...
 import PropTypes from 'prop-types';
 import TextareaWidget from 'react-jsonschema-form/lib/components/widgets/TextareaWidget';
+import { importFromCDN } from '@talend/react-components';
+
+const AceEditor = lazy(() =>
+	importFromCDN({
+		name: 'react-ace',
+		varName: 'ReactAce',
+		version: '6.2.0',
+		path: '/dist/react-ace.min.js',
```
Comment (frassinier) [inline packages/forms/src/deprecated/widgets/CodeWidget/CodeWidget.component.js:11]: Could be neat to use the mapping from DynamicWebpackPlugin instead?

File: packages/forms/src/UIForm/fields/Code/__snapshots__/Code.component.test.js.snap [hunk 5]
```diff
@@ -21,54 +21,39 @@ exports[`Code field should render ace-editor in FieldTemplate 1`] = `
...
+          Loading...
+        </div>
       }
-      setOptions={
-        Object {
-          "enableBasicAutocompletion": true,
-          "enableLiveAutocompletion": true,
-          "enableSnippets": true,
+    >
+      <UNDEFINED
```
Comment (frassinier) [inline packages/forms/src/UIForm/fields/Code/__snapshots__/Code.component.test.js.snap:31]: 😬

File: packages/components/src/index.js [hunk 6]
```diff
@@ -22,6 +22,7 @@ import CollapsiblePanel from './CollapsiblePanel';
 import ConfirmDialog from './ConfirmDialog';
 import Datalist from './Datalist';
 import { ModelViewer, RecordsViewer } from './DataViewer';
+import { importFromCDN } from './importFromCDN';
```
Comment (frassinier) [inline packages/components/src/index.js:25]: What do you think about a babel macro instead? Defined in TUI Scripts?

File: packages/components/src/importFromCDN.js (hunk 1 above)
Comment (jmfrancois) [reply packages/components/src/importFromCDN.js:23]: Yes but I have no way to track status of the loading
The only option is to put some console.log at some pace may be ?

File: packages/forms/src/UIForm/fields/Code/__snapshots__/Code.component.test.js.snap (hunk 5 above)
Comment (jmfrancois) [reply packages/forms/src/UIForm/fields/Code/__snapshots__/Code.component.test.js.snap:31]: May be just inject the global var before the tests are runned should fix that

File: packages/components/src/importFromCDN.js (hunk 1 above)
Comment (frassinier) [reply packages/components/src/importFromCDN.js:23]: It could be neat, with a timeout?

File: packages/components/src/index.js (hunk 6 above)
Comment (jmfrancois) [reply packages/components/src/index.js:25]: AFAIK macro are good when you have few lines of code here I am not sure of the qtt

**Comment by jmfrancois** [comment]
super seed by #3692

**Pull request closed without merging**